hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -271,61 +359,130 @@ private Set<String> getDeltaFiles(OmSnapshot fromSnapshot,
+      final PersistentMap<Long, DiffReportEntry> createDiffs =
+          createDiffReportPersistentMap(createDiffColumnFamily);
+      final PersistentMap<Long, DiffReportEntry> modifyDiffs =
+          createDiffReportPersistentMap(modifyDiffColumnFamily);
+
+
+      long deleteCounter = 0L;
+      long renameCounter = 0L;
+      long createCounter = 0L;
+      long modifyCounter = 0L;
+
+      Iterator<Long> objectIdsIterator = objectIDsToCheck.getKeyIterator();
+      while (objectIdsIterator.hasNext()) {
+        Long id = objectIdsIterator.next();
+        /*
+         * This key can be
+         * -> Created after the old snapshot was taken, which means it will be
+         *    missing in oldKeyTable and present in newKeyTable.
+         * -> Deleted after the old snapshot was taken, which means it will be
+         *    present in oldKeyTable and missing in newKeyTable.
+         * -> Modified after the old snapshot was taken, which means it will be
+         *    present in oldKeyTable and present in newKeyTable with same
+         *    Object ID but with different metadata.
+         * -> Renamed after the old snapshot was taken, which means it will be
+         *    present in oldKeyTable and present in newKeyTable but with
+         *    different name and same Object ID.
+         */
+
+        final String oldKeyName = oldObjIdToKeyMap.get(id);
+        final String newKeyName = newObjIdToKeyMap.get(id);
+
+        if (oldKeyName == null && newKeyName == null) {
+          // This cannot happen.
+          throw new IllegalStateException("Old and new key name both are null");
+        } else if (oldKeyName == null) { // Key Created.
+          createDiffs.put(createCounter++,
+              DiffReportEntry.of(DiffType.CREATE, newKeyName));
+        } else if (newKeyName == null) { // Key Deleted.
+          deleteDiffs.put(deleteCounter++,
+              DiffReportEntry.of(DiffType.DELETE, oldKeyName));
+        } else if (oldKeyName.equals(newKeyName)) { // Key modified.
+          modifyDiffs.put(modifyCounter++,
+              DiffReportEntry.of(DiffType.MODIFY, newKeyName));
+        } else { // Key Renamed.
+          renameDiffs.put(renameCounter++,
+              DiffReportEntry.of(DiffType.RENAME, oldKeyName, newKeyName));
+        }
       }
 
-      // Key Created.
-      if (oldKeyName == null) {
-        createDiffs.add(DiffReportEntry.of(DiffType.CREATE, newKeyName));
-        continue;
+      return aggregateDiffReports(deleteDiffs,
+          renameDiffs,
+          createDiffs,
+          modifyDiffs);
+    } finally {
+      if (deleteDiffColumnFamily != null) {
+        rocksDB.dropColumnFamily(deleteDiffColumnFamily);
+        deleteDiffColumnFamily.close();
       }
-
-      // Key Deleted.
-      if (newKeyName == null) {
-        deleteDiffs.add(DiffReportEntry.of(DiffType.DELETE, oldKeyName));
-        continue;
+      if (renameDiffColumnFamily != null) {
```

Comment:
One benefit I see of storing these diffs is if we don't drop them here and persist them with the from and to snapshot Info we can easily use it to store precomputed snapdiffs.